cloudinit/stages.py

```diff
@@ -275,15 +275,36 @@ def read_cfg(self, extra_fns=None):
             self._cfg = self._read_cfg(extra_fns)
 
     def _read_cfg(self, extra_fns):
-        no_cfg_paths = helpers.Paths({}, self.datasource)
+        """read and merge our configuration"""
+        # No config is passed to Paths() here because we don't yet have a
+        # config to pass. We must bootstrap a config to identify
+        # distro-specific run_dir locations. Once we have the run_dir
+        # we re-read our config with a valid Paths() object. This code has to
+        # assume the location of /etc/cloud/cloud.cfg && /etc/cloud/cloud.cfg.d
+
+        inital_config = self._read_bootstrap_cfg(extra_fns, {})
+        paths = inital_config.get("system_info", {}).get("paths", {})
+
+        # run_dir hasn't changed so we can safely return the config
+        if DEFAULT_RUN_DIR == paths.get("run_dir"):
```

Comment:
Catch here if run_dir isn't defined too
```suggestion
        if paths.get("run_dir") in (DEFAULT_RUN_DIR, None):
```